include/Core/UI/Setting/CoreSetting.php

```diff
@@ -212,4 +222,205 @@ public function render_hide_dashboard_widget_settings( $args ) {
+
+		// wp_parse_args won't merge nested array keys. So add the key here if it is not set.
+		if ( ! array_key_exists( 'notify', $db_size_notification_data ) ) {
+			$db_size_notification_data['notify'] = false;
+		}
+		if ( ! array_key_exists( 'log_threshold_met', $db_size_notification_data ) ) {
+			$db_size_notification_data['log_threshold_met'] = false;
+		}
+
+		return $db_size_notification_data;
+	}
+
+	/**
+	 * Triggers the Cron to notify admin via email.
+	 *
+	 * Email is triggered only when the threshold setting is met.
+	 *
+	 * @since 2.3.0
+	 */
+	public function verify_email_log_threshold() {
+		$cron_hook = 'el_trigger_notify_email_when_log_threshold_met';
+		if ( ! wp_next_scheduled( $cron_hook ) ) {
+			wp_schedule_event( time(), 'hourly', $cron_hook );
+		}
+	}
+
+	/**
+	 * Send the Threshold notification email to the admin.
+	 *
+	 * @since 2.3.0
+	 */
+	public function trigger_threshold_met_notification_email() {
+		$email_log  = email_log();
+		$logs_count = intval( $email_log->table_manager->get_logs_count() );
+
+		$setting_data = $this->get_value();
+
+		// Return early.
+		if ( ! array_key_exists( 'db_size_notification', $setting_data ) ) {
+			return;
+		}
+
+		$db_size_notification_key  = 'db_size_notification';
+		$db_size_notification_data = $setting_data[ $db_size_notification_key ];
+
+		// Return early.
+		if ( ! array_key_exists( 'logs_threshold', $db_size_notification_data ) ||
+		     ! array_key_exists( 'admin_email', $db_size_notification_data ) ||
+		     ! array_key_exists( 'notify', $db_size_notification_data ) ) {
+			return;
+		}
+
+		$is_notification_enabled = $db_size_notification_data['notify'];
+		$admin_email             = $db_size_notification_data['admin_email'];
+		$logs_threshold          = intval( $db_size_notification_data['logs_threshold'] );
+		$logs_threshold_met      = $db_size_notification_data['log_threshold_met'];
+
+		// Ideally threshold cannot be 0. Also, skip sending email if it is already sent.
+		if ( $logs_threshold === 0 || $logs_threshold_met === true ) {
+			return;
+		}
+
+		if ( $is_notification_enabled && is_email( $admin_email ) && ( $logs_count >= $logs_threshold ) ) {
+			$subject = sprintf( __( 'Email Log Plugin: Your log threshold of %s has been met', 'email-log' ),
+				$logs_threshold );
+			$message = <<<EOT
+<p>This email is generated by the Email Log plugin.</p>
+<p>Your log threshold of $logs_threshold has been met. You may manually delete the logs to keep your database table in size.</p>
+<p>Also, consider using our <a href="https://wpemaillog.com/addons/auto-delete-logs/">Auto Delete Logs</a> plugin to delete the logs automatically.</p>
+EOT;
+			$headers = array( 'Content-Type: text/html; charset=UTF-8' );
+
+			wp_mail( $admin_email, $subject, $message, $headers );
+
+			$setting_data[ $db_size_notification_key ]['log_threshold_met'] = true;
+			\update_option( $this->section->option_name, $setting_data );
+
+			$this->register_threshold_met_admin_notice();
```

Comment:
This notice should be outside the `admin email` and `notification_enabled` checks.

Otherwise the user will not see the notification even if the email is not valid.

Reply:
Fixed.